MigrationTutorial/Blog-Post.md

```diff
@@ -0,0 +1,660 @@
+
+To summarise, in this migration you just saw how to:
+1. change the backing value of a model's property (from a string to an enum) *16)*
+1. rename a property of a model *17)*
+1. move the Primary Key (PKs) from a model's property to another one and handle the deletion of duplicated PKs *18)*
+
+</br>
+
+---
+
+#### *(11) V2 Migration function*
+```cs
+public static void DoMigrate(Migration migration, ulong oldSchemaVersion)
+{
+    var newEmployees = migration.NewRealm.All<Employee>(); // 14)
+    var oldEmployees = migration.OldRealm.DynamicApi.All("Employee");   // 15)
+
+    // 16) change Gender property from string to enum
+    for (var i = 0; i < newEmployees.Count(); i++)
+    {
+        var newEmployee = newEmployees.ElementAt(i);
+        var oldEmployee = oldEmployees.ElementAt(i);
+        if (string.Equals(oldEmployee.Gender, "female", StringComparison.OrdinalIgnoreCase))
+        {
+            newEmployee.Gender = Gender.Female;
+        }
+        else if (string.Equals(oldEmployee.Gender, "male", StringComparison.OrdinalIgnoreCase))
+        {
+            newEmployee.Gender = Gender.Male;
+        }
+        else
+        {
+            newEmployee.Gender = Gender.Other;
+        }
+    }
+
+    // 17) rename Consumable.Price to Consumable.LastPurchasedPrice
+    migration.RenameProperty(nameof(Consumable), "Price", nameof(Consumable.LastPurchasedPrice));
+
+    // 18) remove duplicated Consumable to accommodate ProductId to become the new primary key
+    var newConsumables = migration.NewRealm.All<Consumable>();
+    var distinctConsumableId = new HashSet<string>();
+    var consumableToDelete = new List<Consumable>();
+
+    for (var i = 0; i < newConsumables.Count(); i++)
+    {
+        var currConsumable = newConsumables.ElementAt(i);
+
+        // remove duplicates since ProductId is the new PrimaryKey
+        if (distinctConsumableId.Contains(currConsumable.ProductId))
+        {
+            consumableToDelete.Add(currConsumable);
+        }
+        else
+        {
+            distinctConsumableId.Add(currConsumable.ProductId);
+        }
+    }
+
+    consumableToDelete.ForEach(x => migration.NewRealm.Remove(x));
+}
+```
+
+---
+
+</br>
+
+### V2 to V3
+
+#### V3 Models
+After some time, given that the company keeps expanding, a new `Department` called `Workshop` is introduced. This `Department` is now in charge of maintaining `MachineryAndTool`s [(12)](#12-model-v3machineryandtool). Because of this addition the company reclassifies two `Consumable`, `Brush` *19)* and `GlueHolder` *20)*, as `MachineryAndTool` since it is more environmentally friendly and economically advantageous to properly care for them. 
```

Comment:
Changed `Consumable` to `Consumable`s.

```suggestion
After some time, given that the company keeps expanding, a new `Department` called `Workshop` is introduced. This `Department` is now in charge of maintaining `MachineryAndTool`s [(12)](#12-model-v3machineryandtool). Because of this addition the company reclassifies two `Consumable`s, `Brush` *19)* and `GlueHolder` *20)*, as `MachineryAndTool` since it is more environmentally friendly and economically advantageous to properly care for them. 
```